include/depthimage_to_laserscan/DepthImageToLaserScan.h

```diff
@@ -181,31 +187,81 @@ namespace depthimage_to_laserscan
       const T* depth_row = reinterpret_cast<const T*>(&depth_msg->data[0]);
       int row_step = depth_msg->step / sizeof(T);
 
-      int offset = (int)(cam_model.cy()-scan_height/2);
+      // offset by certain amonut since the bottom 150 row are mostly floor (noisy data)
+      int offset = (int)(center_y - scan_height/2 + OFFSET_BOTTOM_ROW/2);
       depth_row += offset*row_step; // Offset to center of image
 
+      // listen to transform only once to avoid overhead
+      tf::StampedTransform transform;
+      try{
+        listener_.lookupTransform("/map", "/camera_depth_optical_frame", ros::Time(0), transform);
+      } catch (tf::TransformException &ex){
+        ROS_ERROR("%s", ex.what());
+        return;
+      }
+
+      // store specific element we want to use
+      double tf_basis_2_0 = transform.getBasis()[2][0];
+      double tf_basis_2_1 = transform.getBasis()[2][1];
+      double tf_basis_2_2 = transform.getBasis()[2][2];
+      double tf_origin_z = transform.getOrigin().z();
+
       for(int v = offset; v < offset+scan_height_; v++, depth_row += row_step){
-		for (int u = 0; u < (int)depth_msg->width; u++) // Loop over each pixel in row
-		{	
-		  T depth = depth_row[u];
-		  
-		  double r = depth; // Assign to pass through NaNs and Infs
-		  double th = -atan2((double)(u - center_x) * constant_x, unit_scaling); // Atan2(x, z), but depth divides out
-		  int index = (th - scan_msg->angle_min) / scan_msg->angle_increment;
+	for (int u = 0; u < (int)depth_msg->width; u++) // Loop over each pixel in row
+	{	
+	  T depth = depth_row[u];
 		  
-		  if (depthimage_to_laserscan::DepthTraits<T>::valid(depth)){ // Not NaN or Inf
-		    // Calculate in XYZ
-		    double x = (u - center_x) * depth * constant_x;
-		    double z = depthimage_to_laserscan::DepthTraits<T>::toMeters(depth);
-		    
-		    // Calculate actual distance
-		    r = sqrt(pow(x, 2.0) + pow(z, 2.0));
-		  }
+	  double r = depth; // Assign to pass through NaNs and Infs
+          // Because the camera is flipped, the theta should also be flipped
+	  double th = atan2((double)(u - center_x) * constant_x, unit_scaling); // Atan2(x, z), but depth divides out
```

Comment:
The map => laser_frame should handle this.  We should look into why this is necessary or how to use the TF.  This way when/if we fix the orientation this will also be fixed.